src/main/sensors/gyroanalyse.c

```diff
@@ -93,27 +91,19 @@ void initGyroData(void)
     }
 }
 
-static inline int fftFreqToBin(int freq)
-{
-    return ((FFT_WINDOW_SIZE / 2 - 1) * freq) / (fftMaxFreq);
-}
-
 void gyroDataAnalyseInit(uint32_t targetLooptimeUs)
 {
     // initialise even if FEATURE_DYNAMIC_FILTER not set, since it may be set later
-    samplingFrequency = 1000000 / targetLooptimeUs;
+    const uint16_t samplingFrequency = 1000000 / targetLooptimeUs;
     fftSamplingScale = samplingFrequency / FFT_SAMPLING_RATE;
-    fftMaxFreq = FFT_SAMPLING_RATE / 2;
-    fftBinCount = fftFreqToBin(fftMaxFreq) + 1;
-    fftResolution = FFT_SAMPLING_RATE / FFT_WINDOW_SIZE;
     arm_rfft_fast_init_f32(&fftInstance, FFT_WINDOW_SIZE);
 
     initGyroData();
     initHanning();
 
     // recalculation of filters takes 4 calls per axis => each filter gets updated every 3 * 4 = 12 calls
     // at 4khz gyro loop rate this means 4khz / 4 / 3 = 333Hz => update every 3ms
-    float looptime = targetLooptimeUs * 4 * 3;
+    float looptime = 1000000 / FFT_SAMPLING_RATE + targetLooptimeUs * 4 * 3;
```

Comment:
Idea is that FFT calculation steps are triggered after the new sample is acquired (1kHz) so the new center frequency for each axis is recalculated with period of 1KHz + steps needed fo calculation.

Reply:
There is a certain lag because of the steps needed, but this must not be confused with the actual time between updates.
Especially at >= 16kHz the new formula is incorrect.
At 16kHz you get targetLooptimeUs == 1/16000 == 62.5us.
If you put this into the new formula you get 1000000µs / 1000Hz + 62.5µs * 4 * 3 == 1750µs² (nope)
Now you initialize the filter for center frequency with a looptime of 1750µs == 571,43Hz.
The correct parameter here would be 1000µs or 1kHz. As this is the rate at which the filter gets updated, no matter how many steps it takes.

Reply:
Now I am getting little bit confused :) 

So the correct value is:
gyro rate >= 16KHz: 1KHz update freq
gyro rate < 16KHz:  old formula (even if we are sampling still at 1KHz) ?

Reply:
Exactly.
Samples for FFT come in at 1kHz, but this is decoupled of the processing at 4 / 8 / 16 / 32kHz.
At 16kHz you can calculate 16 steps before a new sample arrives. As we only need 12, all axis are updated on time and the update rate therefore is 1kHz.

At 8kHz you can only update 2 of 3 axis before a new sample arrives, so that's 666Hz / 1.5ms looptime.
At 4kHz only one axis can be updated. The next axis will work on the next sample. So each axis has 333Hz update rate.

Reply:
Cool, thanks, I'll incorporate the change.